src/ComposerRequireChecker/NodeVisitor/DefinedSymbolCollector.php

```diff
@@ -48,7 +48,7 @@ public function enterNode(Node $node)
 
     private function recordClassDefinition(Node $node)
     {
-        if ($node instanceof Node\Stmt\Class_) {
+        if ($node instanceof Node\Stmt\Class_ && !$node->isAnonymous()) {
```

Comment:
I've tweaked the Travis config to test the lowest stable versions of the deps. on PHP7 and run the default on PHP 7.1. As to cover this case and future ones as well :)